test/ng/qSpec.js

```diff
@@ -2167,45 +2165,69 @@ describe('q', function() {
 
 
   describe('when exceptionHandler is called', function() {
-    it('should log an unhandled rejected promise', function() {
-      var defer = q.defer();
-      defer.reject('foo');
-      mockNextTick.flush();
-      expect(exceptionHandlerStr()).toBe('Possibly unhandled rejection: foo');
-    });
+    var exceptionEg = new Error('Fail');
+    var exceptionStr = toDebugString(exceptionEg);
+    var fixtures = [
+      {
+        type: 'exception',
+        value: exceptionEg,
+        expected: {
+          exception: exceptionEg,
+          reason: 'Possibly unhandled rejection: ' + exceptionStr
+        }
+      },
+      {
+        type: 'plain value',
+        value: 'foo',
+        expected: {
+          reason: 'Possibly unhandled rejection: foo'
+        }
+      }
+    ];
+    forEach(fixtures, function(fixture) {
+      var type = fixture.type;
+      var value = fixture.value;
+      var expected = fixture.expected;
+      it('should log an unhandled' + type + ' rejected promise', function() {
```

Comment:
Instead of repeating `type` in all `it` blocks, create a `describe block for each fixture and put `type` in its description.